Userland/Libraries/LibJS/Runtime/ArrayBufferConstructor.cpp

```diff
@@ -46,10 +46,34 @@ ThrowCompletionOr<Value> ArrayBufferConstructor::call()
     return vm.throw_completion<TypeError>(global_object(), ErrorType::ConstructorWithoutNew, vm.names.ArrayBuffer);
 }
 
+// 1.1.6 GetArrayBufferMaxByteLengthOption ( options ), https://tc39.es/proposal-resizablearraybuffer/#sec-getarraybuffermaxbytelengthoption
+ThrowCompletionOr<Optional<size_t>> ArrayBufferConstructor::get_array_buffer_max_byte_length_option(Value options)
+{
+    // 1. If Type(options) is not Object, return empty.
+    if (!options.is_object())
+        return Optional<size_t>();
+
+    // 2. Let maxByteLength be ? Get(options, "maxByteLength").
+    auto max_byte_length = TRY(options.get(global_object(), "maxByteLength"));
```

Comment:
Instead of taking the constructor's global object here, we need to pass one to the function for cross-realm JS to work properly, and then use that here.